### ISSUE TITLE
Add the ability to reset the optimizer

### ISSUE THREAD

**Issue by wewark** [OpenNMT/OpenNMT-py#970]
I was training using sgt with 0.5 lr decay until it reached almost 0 and stopped learning. I wanted to continue training the same model but using adam with 0.001 lr instead. But it just kept showing 0 as learning rate.

I found a workaround in #768:
> @abaheti95
> if you want to init learning rate when training from a trained model ,you can hack in the source.
> tips:
> just like lua language ,you can add args `-continue` , befor that ,you should change opt.py.
> also, if you want to change laearning rate , just modify the code in `onmt/utils/optimizers.py` named `build_optim`, when load parmas from trained model , you can hack in , and change the rate from the opt.learning_rate.

But I think an argument like `-reset_optim` would be a good idea.

**Comment by vince62s** [comment]
feel free to submit a PR but there are many different situations regarding "resetting" the optimizer.
